docs/rules/no-unused-expressions.md

```diff
@@ -59,6 +59,8 @@ delete a.b
 void a
 ```
 
+Strict mode directives are also not considered problems.
```

Comment:
@platinumazure Here is the relevant code from this rule:

```js
function looksLikeDirective(node) {
    return node.type === "ExpressionStatement" &&
        node.expression.type === "Literal" && typeof node.expression.value === "string";
}
```

It is consistent with the way that Douglas Crockford describes strict mode directive in a classic blog article 
http://yuiblog.com/blog/2010/12/14/strict-mode-is-coming-to-town/


“It has the form of a useless string literal statement, so it will be ignored by ES3 systems.”

When I first read this rule page a few months ago, the question did come to my mind: what about a `"use strict"` string? Looking for an explicit answer distracted me for a moment. Sure, I soon decided that the rule would have to ignore them.

So what y’all think about a sentence at the end of Rule Details? See lines-around-comment and no-implicit-global.

Here is a first draft for comment:

This rule does not apply to directives (which are in the form of literal string expressions) such as `"use strict"`.